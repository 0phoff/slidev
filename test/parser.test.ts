import { resolve, basename } from 'path'
import fg from 'fast-glob'
import { prettify, load, stringify } from '../packages/vite-slides/node'

describe('md parser', () => {
  const files = fg.sync('*.md', {
    cwd: resolve(__dirname, 'fixtures/markdown'),
    absolute: true,
  })

  for (const file of files) {
    it(basename(file), async() => {
      const data = await load(file)

      expect(stringify(data).trim()).toEqual(data.raw.trim())

      prettify(data)

      expect(stringify(data)).toMatchSnapshot('formatted')
    })
  }
})
