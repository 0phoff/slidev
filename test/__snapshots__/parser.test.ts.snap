// Vitest Snapshot v1

exports[`md parser > frontmatter.md > config 1`] = `
{
  "aspectRatio": 1.7777777777777777,
  "canvasWidth": 980,
  "colorSchema": "auto",
  "download": false,
  "drawings": {
    "enabled": true,
    "persist": false,
    "presenterOnly": false,
    "syncAll": true,
  },
  "favicon": "https://cdn.jsdelivr.net/gh/slidevjs/slidev/assets/favicon.png",
  "fonts": {
    "italic": false,
    "local": [],
    "mono": [
      "\\"Fira Code\\"",
      "ui-monospace",
      "SFMono-Regular",
      "Menlo",
      "Monaco",
      "Consolas",
      "\\"Liberation Mono\\"",
      "\\"Courier New\\"",
      "monospace",
    ],
    "provider": "google",
    "sans": [
      "\\"Roboto\\"",
      "\\"Lato\\"",
      "ui-sans-serif",
      "system-ui",
      "-apple-system",
      "BlinkMacSystemFont",
      "\\"Segoe UI\\"",
      "Roboto",
      "\\"Helvetica Neue\\"",
      "Arial",
      "\\"Noto Sans\\"",
      "sans-serif",
      "\\"Apple Color Emoji\\"",
      "\\"Segoe UI Emoji\\"",
      "\\"Segoe UI Symbol\\"",
      "\\"Noto Color Emoji\\"",
    ],
    "serif": [
      "\\"Mate SC\\"",
      "ui-serif",
      "Georgia",
      "Cambria",
      "\\"Times New Roman\\"",
      "Times",
      "serif",
    ],
    "webfonts": [
      "Roboto",
      "Lato",
      "Mate SC",
      "Fira Code",
    ],
    "weights": [
      "200",
      "400",
      "600",
    ],
  },
  "highlighter": "prism",
  "info": false,
  "layout": "cover",
  "lineNumbers": false,
  "monaco": "dev",
  "plantUmlServer": "https://www.plantuml.com/plantuml",
  "remoteAssets": false,
  "routerMode": "history",
  "selectable": false,
  "theme": "default",
  "themeConfig": {},
  "title": "Hi",
  "titleTemplate": "%s - Slidev",
}
`;

exports[`md parser > frontmatter.md > features 1`] = `
{
  "katex": false,
  "mermaid": false,
  "monaco": false,
  "tweet": false,
}
`;

exports[`md parser > frontmatter.md > slides 1`] = `
[
  {
    "content": "
# Hi
",
    "end": 8,
    "frontmatter": {
      "fonts": {
        "mono": "Fira Code",
        "sans": "Roboto, Lato",
        "serif": "Mate SC",
      },
      "layout": "cover",
      "title": "Hi",
    },
    "index": 0,
    "level": 1,
    "note": undefined,
    "raw": "---
layout: cover
fonts:
  sans: Roboto, Lato
  serif: Mate SC
  mono: Fira Code
title: Hi
---

# Hi

",
    "start": 0,
    "title": "Hi",
  },
  {
    "content": "
# Hello
",
    "end": 16,
    "frontmatter": {
      "layout": "center",
      "meta": {
        "duration": 12,
        "title": "FooBar",
      },
    },
    "index": 1,
    "level": 1,
    "note": "This is note",
    "raw": "---
meta:
  title: FooBar
  duration: 12
layout: center
---

# Hello

<!--
This is note
-->
",
    "start": 8,
    "title": "Hello",
  },
  {
    "content": "
# Morning
",
    "end": 19,
    "frontmatter": {},
    "index": 2,
    "level": 1,
    "note": undefined,
    "raw": "
# Morning

",
    "start": 17,
    "title": "Morning",
  },
  {
    "content": "
<!-- This is not note -->
Hey
",
    "end": 26,
    "frontmatter": {
      "layout": "text",
    },
    "index": 3,
    "level": undefined,
    "note": "This is note",
    "raw": "---
layout: text
---

<!-- This is not note -->
Hey

<!--
This is note
-->
",
    "start": 19,
    "title": undefined,
  },
  {
    "content": "
\`\`\`md
---
this should be treated as code block
---
----
\`\`\`
",
    "end": 35,
    "frontmatter": {},
    "index": 4,
    "level": undefined,
    "note": undefined,
    "raw": "
\`\`\`md
---
this should be treated as code block
---
----
\`\`\`

",
    "start": 27,
    "title": undefined,
  },
]
`;

exports[`md parser > minimal.md > config 1`] = `
{
  "aspectRatio": 1.7777777777777777,
  "canvasWidth": 980,
  "colorSchema": "auto",
  "download": false,
  "drawings": {
    "enabled": true,
    "persist": false,
    "presenterOnly": false,
    "syncAll": true,
  },
  "favicon": "https://cdn.jsdelivr.net/gh/slidevjs/slidev/assets/favicon.png",
  "fonts": {
    "italic": false,
    "local": [],
    "mono": [
      "ui-monospace",
      "SFMono-Regular",
      "Menlo",
      "Monaco",
      "Consolas",
      "\\"Liberation Mono\\"",
      "\\"Courier New\\"",
      "monospace",
    ],
    "provider": "google",
    "sans": [
      "ui-sans-serif",
      "system-ui",
      "-apple-system",
      "BlinkMacSystemFont",
      "\\"Segoe UI\\"",
      "Roboto",
      "\\"Helvetica Neue\\"",
      "Arial",
      "\\"Noto Sans\\"",
      "sans-serif",
      "\\"Apple Color Emoji\\"",
      "\\"Segoe UI Emoji\\"",
      "\\"Segoe UI Symbol\\"",
      "\\"Noto Color Emoji\\"",
    ],
    "serif": [
      "ui-serif",
      "Georgia",
      "Cambria",
      "\\"Times New Roman\\"",
      "Times",
      "serif",
    ],
    "webfonts": [],
    "weights": [
      "200",
      "400",
      "600",
    ],
  },
  "highlighter": "prism",
  "info": false,
  "lineNumbers": false,
  "monaco": "dev",
  "plantUmlServer": "https://www.plantuml.com/plantuml",
  "remoteAssets": false,
  "routerMode": "history",
  "selectable": false,
  "theme": "default",
  "themeConfig": {},
  "title": "H1",
  "titleTemplate": "%s - Slidev",
}
`;

exports[`md parser > minimal.md > features 1`] = `
{
  "katex": false,
  "mermaid": false,
  "monaco": false,
  "tweet": false,
}
`;

exports[`md parser > minimal.md > slides 1`] = `
[
  {
    "content": "
# H1
## H2
### H3

Sample Text

\`\`\`ts
console.log('Hello World')
\`\`\`
",
    "end": 10,
    "frontmatter": {
      "title": "H1",
    },
    "index": 0,
    "level": 1,
    "note": undefined,
    "raw": "---
title: H1
---

# H1
## H2
### H3

Sample Text

\`\`\`ts
console.log('Hello World')
\`\`\`

",
    "start": 0,
    "title": "H1",
  },
  {
    "content": "
# Hello

- Hello
- Hi
- Hey
- Yo
",
    "end": 18,
    "frontmatter": {},
    "index": 1,
    "level": 1,
    "note": undefined,
    "raw": "
# Hello

- Hello
- Hi
- Hey
- Yo

",
    "start": 11,
    "title": "Hello",
  },
  {
    "content": "
Nice to meet you
",
    "end": 22,
    "frontmatter": {},
    "index": 2,
    "level": undefined,
    "note": undefined,
    "raw": "
Nice to meet you

",
    "start": 19,
    "title": undefined,
  },
]
`;

exports[`md parser > multi-entries.md > config 1`] = `
{
  "aspectRatio": 1.7777777777777777,
  "canvasWidth": 980,
  "colorSchema": "auto",
  "download": false,
  "drawings": {
    "enabled": true,
    "persist": false,
    "presenterOnly": false,
    "syncAll": true,
  },
  "favicon": "https://cdn.jsdelivr.net/gh/slidevjs/slidev/assets/favicon.png",
  "fonts": {
    "italic": false,
    "local": [],
    "mono": [
      "ui-monospace",
      "SFMono-Regular",
      "Menlo",
      "Monaco",
      "Consolas",
      "\\"Liberation Mono\\"",
      "\\"Courier New\\"",
      "monospace",
    ],
    "provider": "google",
    "sans": [
      "ui-sans-serif",
      "system-ui",
      "-apple-system",
      "BlinkMacSystemFont",
      "\\"Segoe UI\\"",
      "Roboto",
      "\\"Helvetica Neue\\"",
      "Arial",
      "\\"Noto Sans\\"",
      "sans-serif",
      "\\"Apple Color Emoji\\"",
      "\\"Segoe UI Emoji\\"",
      "\\"Segoe UI Symbol\\"",
      "\\"Noto Color Emoji\\"",
    ],
    "serif": [
      "ui-serif",
      "Georgia",
      "Cambria",
      "\\"Times New Roman\\"",
      "Times",
      "serif",
    ],
    "webfonts": [],
    "weights": [
      "200",
      "400",
      "600",
    ],
  },
  "highlighter": "prism",
  "info": false,
  "lineNumbers": false,
  "monaco": "dev",
  "plantUmlServer": "https://www.plantuml.com/plantuml",
  "remoteAssets": false,
  "routerMode": "history",
  "selectable": false,
  "src": "sub/page1.md",
  "theme": "default",
  "themeConfig": {},
  "title": undefined,
  "titleTemplate": "%s - Slidev",
}
`;

exports[`md parser > multi-entries.md > features 1`] = `
{
  "katex": true,
  "mermaid": false,
  "monaco": false,
  "tweet": true,
}
`;

exports[`md parser > multi-entries.md > slides 1`] = `
[
  {
    "content": "
# Page 1
",
    "end": 4,
    "frontmatter": {
      "src": "sub/page1.md",
      "title": undefined,
    },
    "index": 0,
    "inline": {
      "content": "",
      "end": 4,
      "frontmatter": {
        "src": "sub/page1.md",
        "title": undefined,
      },
      "index": 0,
      "level": undefined,
      "note": undefined,
      "raw": "---
src: sub/page1.md
---
",
      "start": 0,
      "title": undefined,
    },
    "level": 1,
    "note": undefined,
    "raw": "---
src: sub/page1.md
---

# Page 1

",
    "source": {
      "content": "# Page 1",
      "frontmatter": {},
      "level": 1,
      "note": undefined,
      "raw": "# Page 1
",
      "title": "Page 1",
    },
    "start": 0,
    "title": "Page 1",
  },
  {
    "content": "
# Page 2

<Tweet />
",
    "end": 9,
    "frontmatter": {
      "background": "https://sli.dev/demo-cover.png",
      "layout": "cover",
      "src": "sub/page2.md",
    },
    "index": 1,
    "inline": {
      "content": "",
      "end": 9,
      "frontmatter": {
        "background": "https://sli.dev/demo-cover.png",
        "src": "sub/page2.md",
      },
      "index": 1,
      "level": undefined,
      "note": undefined,
      "raw": "---
src: sub/page2.md
background: https://sli.dev/demo-cover.png
---
",
      "start": 4,
      "title": undefined,
    },
    "level": 1,
    "note": undefined,
    "raw": "---
layout: cover
src: sub/page2.md
background: https://sli.dev/demo-cover.png
---

# Page 2

<Tweet />

",
    "source": {
      "content": "# Page 2

<Tweet />",
      "frontmatter": {
        "layout": "cover",
      },
      "level": 1,
      "note": undefined,
      "raw": "---
layout: cover
---

# Page 2

<Tweet />
",
      "title": "Page 2",
    },
    "start": 4,
    "title": "Page 2",
  },
  {
    "content": "
# Inline Page

$x+2$
",
    "end": 15,
    "frontmatter": {},
    "index": 2,
    "level": 1,
    "note": undefined,
    "raw": "
# Inline Page

$x+2$

",
    "start": 10,
    "title": "Inline Page",
  },
]
`;
